libinteract/libinteract.py

```diff
@@ -866,156 +867,140 @@ def calc_cg_fullmatrix(identifiers, idxs, percmat, perco):
         if chain1 == chain2:
-            warnstr = f"No intrachain contacts found in chain {chain1}"
-            log.warning(warnstr)
+            log.warning(f"No intrachain contacts found in chain {chain1}")
         else:
             warnstr = f"No interchain contacts found between chains " \
                       f"{chain1} and {chain2}"
             log.warning(warnstr)
+        return None
     else:
-        output = filtered_array
-    return output
+        return filtered_rows
 
 
-def create_table_list(contact_list, hb=False):
-    """Takes in a list of tuples and returns a list of arrays where
-    the first array contains all contacts followed by intrachain 
-    contants split by chain (if present) and interchain contacts split 
-    by chain (if present).
+def create_table_dict(table, hb=False):
+    """Takes in a single table (list of tuples) and returns a dictionary
+    of tables (arrays). Each key in this dictionary represent whether the 
+    table contains all/intrachain/interchain contacts.
     """
 
     # Convert to array and keep transpose for selection
-    array = np.array(contact_list)
-    array_T = array.T
-    # Initialize output
-    array_list = [array]
-    # Find unique chains in the array
-    chains = np.unique(array_T[0])
+    table_rows = np.array(table)
+    table_cols = table_rows.T
+    # Initialize output dictionary of tables
+    table_dict = {"all": table_rows}
     # Choose which column has the second residue
-    if hb:
-        sec_res = 4
-    else:
-        sec_res = 3
+    sec_res = 4 if hb else 3
+    # Find unique chains in the table
+    chains = np.unique(np.concatenate((table_cols[0], table_cols[sec_res])))
     # If multiple chains are present, split the contacts by chain
     if len(chains) > 1:
 	# For each chain, add the nodes that are only in contact with the same chain
         for chain in chains:
-            filtered_array = filter_by_chain(chain, chain, sec_res, \
-                                            array, array_T)
-            if filtered_array is not None:
-                array_list.append(filtered_array)
+            filtered_rows = filter_by_chain(chain, chain, sec_res, table_rows)
+            if filtered_rows is not None:
+                table_dict[chain] = filtered_rows
         # Create vector of all nodes that are in contact with different chains
-        logical_vector = array_T[0] != array_T[sec_res]
-        if logical_vector.sum == 0:
-            warnstr = "No interchain contacts found" 
-            log.warning(warnstr)
+        logical_vector = table_cols[0] != table_cols[sec_res]
+        if logical_vector.sum() == 0:
+            log.warning("No interchain contacts found")
         else:
             # For all combinations of different chains, find nodes that are in
             # contact with different chains
             for chain1, chain2 in itertools.combinations(chains, 2):
                 # Check both combinations (e.g. A, B and B, A)
```

Comment:
Maybe this logic (i.e. considering both (A,B) and (B,A) pairs) should actually be in the `filter_by_chain` function. In the end we don't ever want to distinguish between the two and so they should be get ridden of as soon as possible. I think it makes sense logically. 

For the dictionary keys, you can either consider to sort the letters before using them or to use `FrozenSet`s instead of tuples which is maybe the most robust option (they are immutable, hashable objects which don't have an order).

Reply:
I made the changes and used a sorted tuple. 

I tried using `FrozenSet` but had issues when I tried to extract the chain names from the set when I named the files. I tried using `next(iter(key))` to cycle through the set elements but it resulted in the wrong order (B-A.csv instead of A-B.csv).